### PR TITLE
Fix lambdify of Min when using 'numpy' module (fixes gh-10375)

### LINKED ISSUE #10375: lambdify on sympy.Min does not work with NumPy
```
IPython console for SymPy 0.7.7.dev (Python 3.4.3-64-bit) (ground types: python)

These commands were executed:
>>> from __future__ import division
>>> from sympy import *
>>> x, y, z, t = symbols('x y z t')
>>> k, m, n = symbols('k m n', integer=True)
>>> f, g, h = symbols('f g h', cls=Function)
>>> init_printing()

Documentation can be found at http://docs.sympy.org/dev

In [1]: import numpy as np

In [2]: np.__version__
Out[2]: '1.10.1'

In [3]: m = Min(x, y, z)

In [4]: l = lambdify([x,y,z], m)

In [5]: l(1, 2, 3)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/home/bjorn/.local/lib/python3.4/site-packages/numpy/core/fromnumeric.py in amin(a, axis, out, keepdims)
   2341         try:
-> 2342             amin = a.min
   2343         except AttributeError:

AttributeError: 'int' object has no attribute 'min'

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
<ipython-input-5-c33036198c2b> in <module>()
----> 1 l(1, 2, 3)

/home/bjorn/.local/lib/python3.4/site-packages/numpy/__init__.py in <lambda>(_Dummy_18, _Dummy_19, _Dummy_20)

/home/bjorn/.local/lib/python3.4/site-packages/numpy/core/fromnumeric.py in amin(a, axis, out, keepdims)
   2343         except AttributeError:
   2344             return _methods._amin(a, axis=axis,
-> 2345                                   out=out, keepdims=keepdims)
   2346         # NOTE: Dropping the keepdims parameter
   2347         return amin(axis=axis, out=out)

/home/bjorn/.local/lib/python3.4/site-packages/numpy/core/_methods.py in _amin(a, axis, out, keepdims)
     27 
     28 def _amin(a, axis=None, out=None, keepdims=False):
---> 29     return umr_minimum(a, axis, None, out, keepdims)
     30 
     31 def _sum(a, axis=None, dtype=None, out=None, keepdims=False):

TypeError: output must be an array
```

whereas using SymPy module for lambdify does work:
```
In [6]: l = lambdify([x,y,z], m, modules=['sympy'])

In [7]: l(1, 2, 3)
Out[7]: 1
```

### PR DESCRIPTION
**Pull request by bjodah**
This should fix gh-10375

### CODE REVIEW

**Comment by bjodah** [comment]
ping @tokencolour @Shekharrajak

File: sympy/printing/lambdarepr.py [hunk 1]
```diff
@@ -157,6 +157,13 @@ def _print_Not(self, expr):
         #     own because StrPrinter doesn't define it.
         return '{0}({1})'.format('logical_not', ','.join(self._print(i) for i in expr.args))
 
+    def _print_Min(self, expr):
+        return '{0}(({1}))'.format('amin', ','.join(self._print(i) for i in expr.args))
```
Comment (Shekharrajak) [inline sympy/printing/lambdarepr.py]: Why did you use this ?Will you explain little?

**Comment by asmeurer** [comment]
+1. We ideally should remove all isinstance checks, and even the translation dictionary, from lambdify in favor of extending the code printers.

**Comment by asmeurer** [comment]
So does lambdify also work with Min and Max when not using numpy?

File: sympy/utilities/tests/test_lambdify.py [hunk 2]
```diff
@@ -578,6 +578,13 @@ def test_issue_2790():
...
 
+
 def test_ITE():
     assert lambdify((x, y, z), ITE(x, y, z))(True, 5, 3) == 5
     assert lambdify((x, y, z), ITE(x, y, z))(False, 5, 3) == 3
+
+
+def test_Min_Max():
+    # see gh-10375
+    assert lambdify((x, y, z), Min(x, y, z))(1, 2, 3) == 1
```
Comment (Shekharrajak) [inline sympy/utilities/tests/test_lambdify.py]: It seems you didn't `import numpy` and using `Min` , that means it is using `sympy.Min` isn't it ?

Comment (Shekharrajak) [inline sympy/utilities/tests/test_lambdify.py]: Okay,if in the user's system `numpy` is installed the `lambdify` add `numpy` is the modules list automatically.it's fine.

**Comment by Shekharrajak** [comment]
+1 .It works fine for all cases(also when numpy is not installed  ).

**Comment by tokencolour** [comment]
Cool solution! I wish one day I am able to give one line solutions!

File: sympy/utilities/tests/test_lambdify.py (hunk 2 above)
Comment (asmeurer) [inline sympy/utilities/tests/test_lambdify.py]: Yes, this test will test something different if numpy is installed vs. if it isn't. Fortunately, Travis has builds for both cases.

**Pull request merged**